src/tools/kata-ctl/src/monitor/http_server.rs

```diff
@@ -0,0 +1,132 @@
+// Copyright 2022-2023 Kata Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use crate::monitor::metrics::get_metrics;
+use crate::sl;
+use anyhow::{anyhow, Result};
+use hyper::body;
+use hyper::service::{make_service_fn, service_fn};
+use hyper::{Body, Method, Request, Response, Server, StatusCode};
+use shim_interface::shim_mgmt::client::MgmtClient;
+use slog::{self, info};
+use std::collections::HashMap;
+use std::net::SocketAddr;
+use std::time::Duration;
+
+const ROOT_URI: &str = "/";
+const METRICS_URI: &str = "/metrics";
+
+async fn handler_mux(req: Request<Body>) -> Result<Response<Body>, hyper::Error> {
+    info!(
+        sl!(),
+        "mgmt-svr(mux): recv req, method: {}, uri: {}",
+        req.method(),
+        req.uri().path()
+    );
+
+    match (req.method(), req.uri().path()) {
+        (&Method::GET, ROOT_URI) => root_uri_handler(req).await,
+        (&Method::GET, METRICS_URI) => metrics_uri_handler(req).await,
+        _ => Ok(not_found_uri_handler(req).await),
+    }
+}
+
+#[tokio::main]
+pub async fn http_server(socket_addr: &str) -> Result<()> {
+    let addr: SocketAddr = socket_addr.parse()?;
+
+    let make_svc =
+        make_service_fn(|_conn| async { Ok::<_, hyper::Error>(service_fn(handler_mux)) });
+
+    let server = Server::bind(&addr).serve(make_svc);
+
+    if let Err(e) = server.await {
+        eprintln!("server error: {}", e);
+    }
+
+    Ok(())
+}
+
+async fn root_uri_handler(_req: Request<Body>) -> Result<Response<Body>, hyper::Error> {
+    Ok(Response::builder()
+        .status(StatusCode::OK)
+        .body(Body::from(
+            r#"Available HTTP endpoints:
+    /metrics : Get metrics from sandboxes.
+"#,
+        ))
+        .unwrap())
+}
+
+async fn metrics_uri_handler(req: Request<Body>) -> Result<Response<Body>, hyper::Error> {
+    let mut response_body = String::new();
+
+    if let Ok(monitor_metrics) = get_metrics().await {
+        response_body += &monitor_metrics;
+    } else {
+        return Ok(Response::builder()
+            .status(StatusCode::INTERNAL_SERVER_ERROR)
+            .body(Body::from("failed to get monitor metrics"))
+            .unwrap());
+    }
+
+    if let Some(uri_query) = req.uri().query() {
+        if let Ok(sandbox_id) = parse_sandbox_id(uri_query) {
+            let mut runtime_metrics_flag = false;
+            let runtime_metrics = get_runtime_metrics(sandbox_id).await;
```

Comment:
How about:
```
match get_runtime_metrics(sandbox_id).await {
	Ok(runtime_metrics) => {...}
	Err(e) => {...}
}
```

I think empty runtime_metrics can be handled to return "empty runtime metrics" in response body. If the empty situation is regarded as an error, it is better handled when we collecting the metrics, rather than handling here.